cmd/lock-main.go

```diff
@@ -0,0 +1,207 @@
+}
+
+// JSON'ified message for scripting.
+func (m lockCmdMessage) JSON() string {
+	msgBytes, e := json.MarshalIndent(m, "", " ")
+	fatalIf(probe.NewError(e), "Unable to marshal into JSON.")
+	return string(msgBytes)
+}
+
+func lock(urlStr string, mode *minio.RetentionMode, validity *uint, unit *minio.ValidityUnit, clearLock bool) error {
+	client, err := newClient(urlStr)
+	if err != nil {
+		fatalIf(err.Trace(), "Cannot parse the provided url.")
+	}
+
+	s3Client, ok := client.(*s3Client)
+	if !ok {
+		fatalIf(errDummy().Trace(), "The provided url doesn't point to a S3 server.")
+	}
+
+	validityStr := func() *string {
+		if validity != nil {
+			unitStr := "d"
+			if *unit == minio.Years {
+				unitStr = "y"
+			}
+			s := fmt.Sprintf("%v%v", *validity, unitStr)
+			return &s
+		}
+
+		return nil
+	}
+
+	if clearLock || mode != nil {
+		err = s3Client.SetObjectLockConfig(mode, validity, unit)
+		fatalIf(err, "Cannot enable object lock configuration on the specified bucket.")
+		printMsg(lockCmdMessage{
+			Enabled:  "Enabled",
+			Mode:     mode,
+			Validity: validityStr(),
+			Status:   "success",
+		})
+	} else {
+		mode, validity, unit, err = s3Client.GetObjectLockConfig()
+		fatalIf(err, "Cannot get object lock configuration on the specified bucket.")
+
+		printMsg(lockCmdMessage{
+			Enabled:  "Enabled",
+			Mode:     mode,
+			Validity: validityStr(),
+			Status:   "success",
+		})
+	}
+
+	return nil
+}
+
+// main for lock command.
+func mainLock(ctx *cli.Context) error {
+	console.SetColor("Mode", color.New(color.FgCyan, color.Bold))
+	console.SetColor("Validity", color.New(color.FgYellow))
+
+	// Parse encryption keys per command.
+	_, err := getEncKeys(ctx)
+	fatalIf(err, "Unable to parse encryption keys.")
+
+	// lock specific flags.
+	clearLock := ctx.Bool("clear")
+
+	args := ctx.Args()
+
+	var urlStr string
+	var mode *minio.RetentionMode
+	var validity *uint
+	var unit *minio.ValidityUnit
+
+	switch l := len(args); l {
+	case 0:
+		cli.ShowCommandHelpAndExit(ctx, "lock", 1)
+
+	case 1:
+		urlStr = args[0]
+
+	case 3:
+		urlStr = args[0]
+		if clearLock {
+			return fmt.Errorf("clear flag must be passed with target alone")
+		}
+
+		m := minio.RetentionMode(args[1])
+		if !m.IsValid() {
+			return fmt.Errorf("invalid retention mode '%v'", m)
+		}
+
+		mode = &m
+
+		validityStr := args[2]
+		unitStr := string(validityStr[len(validityStr)-1])
+
+		validityStr = validityStr[:len(validityStr)-1]
+		ui64, err := strconv.ParseUint(validityStr, 10, 64)
+		if err != nil {
+			return fmt.Errorf("invalid validity '%v'", args[2])
+		}
+		u := uint(ui64)
+		validity = &u
+
+		switch unitStr {
+		case "d", "D":
+			d := minio.Days
+			unit = &d
+		case "y", "Y":
+			y := minio.Years
+			unit = &y
+		default:
+			return fmt.Errorf("invalid validity format '%v'", args[2])
+		}
+	default:
+		return fmt.Errorf("insufficient or too many arguments")
```

Comment:
`cli.ShowCommandHelpAndExit(ctx, "lock", 1)` maybe?  (and drop case 0)

Reply:
done